File: 1 - General/Shruthikaa V/Medium/Daily Temperatures/Solution.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int templen = temperatures.length;
        int[] result = new int[templen];
        Stack<Integer> stack = new Stack<>();
        
        for (int i = 0; i < templen; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                int idx = stack.pop();
                result[idx] = i - idx;
            }
            stack.push(i);
        }
        
        return result;
    }
}

File: 1 - General/Shruthikaa V/Medium/Daily Temperatures/Solution.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        templen = len(temperatures)
        result = [0] * templen
        stack = []
        
        for i in range(templen):
            while stack and temperatures[i] > temperatures[stack[-1]]:
                idx = stack.pop()
                result[idx] = i - idx
            stack.append(i)
        
        return result
